robots/cmd/kubevirt-presubmit-requirer/main.go

```diff
@@ -0,0 +1,182 @@
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/google/go-github/github"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"k8s.io/test-infra/prow/config"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/project-infra/robots/pkg/querier"
+)
+
+const OrgAndRepoForJobConfig = "kubevirt/kubevirt"
+
+type options struct {
+	port int
+
+	dryRun bool
+
+	TokenPath                      string
+	endpoint                       string
+	jobConfigPathKubevirtPresubmit string
+}
+
+func (o *options) Validate() error {
+	if _, err := os.Stat(o.jobConfigPathKubevirtPresubmit); os.IsNotExist(err) {
+		return fmt.Errorf("jobConfigPathKubevirtPresubmit is required: %v", err)
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether the file should get modified or just modifications printed to stdout.")
+	fs.StringVar(&o.TokenPath, "github-token-path", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	fs.StringVar(&o.endpoint, "github-endpoint", "https://api.github.com/", "GitHub's API endpoint (may differ for enterprise).")
+	fs.StringVar(&o.jobConfigPathKubevirtPresubmit, "job-config-path-kubevirt-presubmit", "", "The directory of the kubevirt presubmit job definitions")
+	err := fs.Parse(os.Args[1:])
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to parse args: %v", err))
+		os.Exit(1)
+	}
+	return o
+}
+
+func main() {
+
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	// TODO: Use global option from the prow config.
+	logrus.SetLevel(logrus.DebugLevel)
+	log := logrus.StandardLogger().WithField("robot", "kubevirt-presubmit-requirer")
+
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		log.WithError(err).Error("Invalid arguments provided.")
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
+	var client *github.Client
+	if o.TokenPath == "" {
+		var err error
+		client, err = github.NewEnterpriseClient(o.endpoint, o.endpoint, nil)
+		if err != nil {
+			log.Panicln(err)
+		}
+	} else {
+		token, err := ioutil.ReadFile(o.TokenPath)
+		if err != nil {
+			log.Panicln(err)
+		}
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: string(token)},
+		)
+		client, err = github.NewEnterpriseClient(o.endpoint, o.endpoint, oauth2.NewClient(ctx, ts))
+		if err != nil {
+			log.Panicln(err)
+		}
+	}
+
+	jobConfig, err := config.ReadJobConfig(o.jobConfigPathKubevirtPresubmit)
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	releases, _, err := client.Repositories.ListReleases(ctx, "kubernetes", "kubernetes", nil)
+	if err != nil {
+		log.Panicln(err)
+	}
+	releases = querier.ValidReleases(releases)
+	if len(releases) == 0 {
+		log.Info("No release found, nothing to do.")
+		os.Exit(0)
+	}
+
+	latestReleaseSemver := querier.ParseRelease(releases[0])
+
+	updated := UpdatePresubmitsAlwaysRunAndOptionalFields(&jobConfig, latestReleaseSemver)
+	if !updated && !o.dryRun {
```

Comment:
Why do we check for dryRun here? As I understand it, if nothing is updated then we can exit, no matter the value of dryRun, am I missing something?

Reply:
In case of dryRun being set we just print the changes to stdout, so we don't yet exit here. WDYT?

Reply:
See https://github.com/kubevirt/project-infra/pull/1295/files#diff-3e7ab658330ea7d1202fbfd93c3094b3856651b0501bc4dd3877128e1f3dbfe8R129-R135

Reply:
MAkes sense :+1: